messages/EchoDialog.csx

```diff
@@ -56,30 +44,22 @@ public class EchoDialog : IDialog<object>
         string text = message.Text;
         text = text.Replace("<at>GifBot</at>", "");
         text = text.Trim();
-
-        // Get html response for gif query
-        string html = string.Empty;
-        string url = @"https://api.giphy.com/v1/gifs/search?api_key=" + Environment.GetEnvironmentVariable("GiphyApiKey") + "&limit=9&rating=g&q=" + Uri.EscapeDataString(text);
-
-        HttpWebRequest request = (HttpWebRequest)WebRequest.Create(url);
-        request.AutomaticDecompression = DecompressionMethods.GZip;
-        
-        using (HttpWebResponse response = (HttpWebResponse)request.GetResponse())
-        using (Stream stream = response.GetResponseStream())
-        using (StreamReader reader = new StreamReader(stream))
-        {
-            html = reader.ReadToEnd();
-        }
         
-        // Pick randomly from the list and get the url for the gif
+        // Query GIPHY for the URL of the GIF to respond with
+        httpClient.BaseAddress = new Uri("https://api.giphy.com/v1/gifs/");
+        HttpResponseMessage response = await httpClient.GetAsync("search?api_key=" + Environment.GetEnvironmentVariable("GiphyApiKey") + "&limit=9&rating=g&q=" + Uri.EscapeDataString(text));
+        var json = await response.Content.ReadAsStringAsync();
+
+        // Select a random GIF URL from the first ten
         Random rnd = new Random();
-        JavaScriptSerializer json = new JavaScriptSerializer();
-        var JSONObj = json.Deserialize<GiphyJson>(html);
-        string link = JSONObj.data[rnd.Next(1, 10)].images["original"].url;
-        
+        JObject o = JObject.Parse(json);
+        string link = o.SelectToken("data").Take(10).Select(z => (string)z["images"]["original"]["url"]).OrderBy(x => rnd.Next()).FirstOrDefault();

```

Comment:
nit: make this 9. we only take the top 9 gifs from Giphy, this was an early design decision I made and want to keep haha (limit=9 in the query).

Reply:
Whoops, fixed. I saw `rnd.Next(1,10)` and I just read the `10` part